terraso_backend/apps/project_management/graphql/site_notes.py

```diff
@@ -66,7 +75,7 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         user = info.context.user
         site_note_id = kwargs["id"]
         site_note = cls.get_or_throw(SiteNote, "id", site_note_id)
-        if not user.has_perm(SiteNote.get_perm("update"), site_note):
+        if not check_site_permission(user, "edit_note", Context(site_note=site_note)):
```

Comment:
question: trying to follow the flow of logic here, what i'm getting is:
- we call `check_site_permission` with the `"edit_note"`, and a context with just a `site_note` attached
- then, in the implementation of `check_site_permission` in `permission_matrix`, since the context doesn't have a `site` attached, we check `"edit_note"` against the `SITE_PERMISSION_BASE` matrix
- since `"edit_note"` doesn't exist on that matrix, we always throw an error on this call

what am i missing? same q for line 100

Reply:
The Context dataclass automatically fills in the site for its site note, so we'll be referencing the appropriate site-based permission matrix.